ledger/ledger-on-memory/src/main/scala/com/daml/ledger/on/memory/InMemoryLedgerStateOperations.scala

```diff
@@ -0,0 +1,55 @@
+package com.daml.ledger.on.memory
+
+import com.daml.ledger.on.memory.InMemoryLedgerReaderWriter.Index
+import com.daml.ledger.on.memory.InMemoryState.MutableLog
+import com.daml.ledger.participant.state.kvutils.KVOffset
+import com.daml.ledger.participant.state.kvutils.api.LedgerRecord
+import com.daml.ledger.participant.state.v1.Offset
+import com.daml.ledger.validator.{
+  BatchingLedgerStateOperations,
+  LedgerStateAccess,
+  LedgerStateOperations,
+  TimedLedgerStateOperations
+}
+import com.daml.ledger.validator.LedgerStateOperations.{Key, Value}
+import com.daml.metrics.Metrics
+
+import scala.concurrent.{ExecutionContext, Future}
+
+private[memory] final class InMemoryLedgerStateOperations(
+    log: InMemoryState.MutableLog,
+    state: InMemoryState.MutableState,
+)(implicit executionContext: ExecutionContext)
+    extends BatchingLedgerStateOperations[Index] {
+  import InMemoryLedgerStateOperations.appendEntry
+
+  override def readState(keys: Seq[Key]): Future[Seq[Option[Value]]] =
+    Future.successful(keys.map(state.get))
+
+  override def writeState(keyValuePairs: Seq[(Key, Value)]): Future[Unit] = {
+    state ++= keyValuePairs
+    Future.unit
+  }
+
+  override def appendToLog(key: Key, value: Value): Future[Index] =
+    Future.successful(appendEntry(log, LedgerRecord(_, key, value)))
+}
+
+private[memory] class InMemoryLedgerStateAccess(state: InMemoryState, metrics: Metrics)(
+    implicit executionContext: ExecutionContext)
+    extends LedgerStateAccess[Index] {
+  override def inTransaction[T](body: LedgerStateOperations[Index] => Future[T]): Future[T] =
+    state.write { (log, state) =>
+      body(new TimedLedgerStateOperations(new InMemoryLedgerStateOperations(log, state), metrics))
+    }
+}
```

Comment:
If you're going to split this stuff out, keep splitting.

Reply:
Can you elaborate, please?

Reply:
You moved `InMemoryLedgerStateOperations` out into its own file, but took `InMemoryLedgerStateAccess` with it. I find this confusing. `InMemoryLedgerStateAccess` is a higher-level API, but it's buried in a file named after a lower-level API. I suggest splitting it into its own file.

Reply:
Thanks for the confirmation.